### DISCUSSION TITLE
Cannot handle generic type by v0.16.3

### DISCUSSION THREAD

**Discussion by babyfish-ct** [swaggo/swag#1790] [Q&A]
Hi 

A type is defined as follows:

```
package middleware

type Response struct {
    ...other fields omitted...
    data any
}
```

Add the following comments for the HTTP method in the Controller:

```
// @Success 200 {object} middleware.Response
// @Failure 500 {object} middleware.Response
```

After executing the relevant commands, the OpenAPI was correctly generated, although the result was not what I expected.

To get the expected result, I added a generic to the Response struct:

```
type Response[T any] struct {
    ...other fields omitted...
    data T
}
```

Then I modified the comments in the HTTP function of the Controller:

```
// @Success 200 {object} middleware.Response[int]
// @Failure 500 {object} middleware.Response[string]
```

After executing the relevant commands and regenerating the OpenAPI, an error occurred: "cannot find type definition".

I searched it in documentation, it clearly states that it supports Go generics: https://github.com/swaggo/swag/tree/master?tab=readme-ov-file#how-to-use-generics

Finally, when searching for other issues, several issues occurred during the period when Go added generics, and the problems were caused by the users using versions of the frameworks that were too old. However, my version is the latest v1.16.3, which is far beyond the required version for this feature, yet the problem still persists.

Has anyone encountered this issue before?

**Comment by kittizz** [comment]
I also encountered this issue.

// @Success 201 {object} helper.BaseResponse[dto.TokenDetail] "User authenticated successfully"
:cannot find type definition: helper.BaseResponse[dto.TokenDetail]

type BaseResponse[TData any] struct {
	Data             TData                     `json:"data,omitempty"` 
	Success          bool                      `json:"success"`     
	Code             ResultCode                `json:"code"`         
	Error            string                    `json:"error,omitempty"`
	ValidationErrors []binding.ValidationError `json:"validationErrors,omitempty"`
}

**Comment by fluxxu** [comment]
add ` --parseDependency` to `swag init`
